### PR TITLE
Import: Add staged import process

### PR DESCRIPTION
**Pull request by KarlStraussberger**
1. scan all files
2. create objects for files and directories
3. create layout

Further changes:
- Rework Configuration Mechanism for Enums and Paths
- Allow setting configuration values via command line
- Make virtual folders unique by path
- Reduce overhead of loading data by passing around object references instead of objectID.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
           python-version: "3.x"
       - name: Install conan
         run: |
-          pip install -U --no-cache-dir conan
+          pip install -U --no-cache-dir conan==1.59
```
Comment (gobater) [inline .github/workflows/ci.yml:44]: Not sure if this and further (conan) changes below are related to this feature. If not, for the sake of separation of concerns shall be (whenever possible) extracted into a different PR allowing the changes (if needed) to be integrated as soon as possible into the master -> so other contributions could benefit of it

Comment (KarlStraussberger) [reply .github/workflows/ci.yml:44]: Good point. It's the time issue again.

File: src/content/import_service.cc [hunk 2]
```diff
@@ -570,31 +607,51 @@ std::pair<int, bool> ImportService::addContainerTree(int parentContainerId, cons
...
+            if (!dirKeyValues.empty()) {
+                subTree = fmt::format("{}@{}", tree, fmt::join(dirKeyValues, "@"));
+            } else {
+                subTree = tree;
+            }
+            auto cont = database->findObjectByPath(subTree, DbFileType::Virtual);
+            if (cont && cont->isContainer())
+                containerMap[subTree] = std::dynamic_pointer_cast<CdsContainer>(cont);
+            else
+                containerMap[subTree] = nullptr;
```
Comment (gobater) [inline src/content/import_service.cc:627]: I'm not able to follow all changes in this commit, but I guess this is the key part that creates a unique ID within the Albums virtual container...
Considering the changes in "src/config/config_definition.cc", it seems that the album artist and the year are considered and therefore it will probably solve Issue #1440.
Looking forward to see this "in action".
Is this part only of the "grb" import mode or also the "traditional"?

Comment (KarlStraussberger) [reply src/content/import_service.cc:627]: It will be available for both. I'm still testing in real life.

**Comment by gobater** [comment]
@KarlStraussberger ,
I was thinking about adding a new virtual folder (similar to "all songs") within the albums of the artist to list the albums chronologically. It's quite easy to do in Javascript. Do you think it would make sense also to have such a feature in the native importer?

**Comment by KarlStraussberger** [comment]
> @KarlStraussberger , I was thinking about adding a new virtual folder (similar to "all songs") within the albums of the artist to list the albums chronologically. It's quite easy to do in Javascript. Do you think it would make sense also to have such a feature in the native importer?

That's something for JavaScript. Maybe it can be adapted for the release version

**Comment by whyman** [comment]
Wowzers, this is a beast, nice work.
